### ISSUE TITLE
Pinpoint Endpoint is Always APNS_SANDBOX

### ISSUE THREAD

**Issue by iheffernan** [aws-amplify/amplify-swift#2809]
### Describe the bug

I'm trying to use Amplify with Pinpoint for APNS notifications. My issue is that the ChannelType for the endpoint created using 

```
try await Amplify.Notifications.Push.registerDevice(
   apnsToken: deviceToken
)
```

is always APNS_SANDBOX, whether I'm testing on a local device through Xcode or the app was installed through TestFlight.

It's not entirely clear how the Push Notification plugin is setting the channel type for the Endpoints - whether it's using some internal flag or base on the provisioning profile.

I am using an Xcode Managed Profile Provisioning Profile, not sure how that plays into all of this.

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Delete all user endpoints using aws-cli pinpoint delete-endpoint
2. Launch my iOS project and call registerDevice in didRegisterForRemoteNotificationsWithDeviceToken
3. Use CLI tool to check my Endpoints and see the ChannelType is APNS_SANDBOX (expected)
4. Delete my Endpoints again
5. Delete the app from my device
6. Install the app through TestFlight
7. I see another Endpoint has been created, but the ChannelType is still APNS_SANDBOX, not APNS as I would have expected
```

### Expected behavior

I would expect that when running the app locally through Xcode the ChannelType for created Enpoints is APNS_SANDBOX and that when running versions of the app installed through TestFlight Endpoints will have a ChannelType of APNS.

### Amplify Framework Version

2.5.0

### Amplify Categories

Analytics

### Dependency manager

Swift PM

### Swift version

5.0

### CLI version

N/A - using existing resources

### Xcode version

14.2 (14C18)

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

Yes

### Regression additional context

In the v1 libraries, calling getEscapeHatch on the analytics plugin allowed me to retrieve the currentEndpointProfile and modify as necessary (although, it typically set the ChannelType correctly). In v2, the currentEndpointProfile is no longer available on the escapeHatch, which seems unfortunate.

### Device

iPhone XS

### iOS Version

16.3.1

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by ruisebas** [comment]
Hi @iheffernan, thanks for reaching out.

Currently, Amplify will determine which channel to use based on the `DEBUG` flag:
1. If the flag present, [it will set `isDebug` to `true`](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Internal/Sources/InternalAWSPinpoint/Context/AWSPinpointFactory.swift#L33-L35).
2. When building the `UpdateEndpoint` input, [if `isDebug` is `true` it will use `APNS_SANBOX`, otherwise `APNS`](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Internal/Sources/InternalAWSPinpoint/Endpoint/EndpointClient.swift#L187).

Could you please validate that:
- The `DEBUG` flag is not included for release builds, under `Swift Compiler - Custom Flags`
- You are uploading a release build to TestFlight.

I hope this clarification in behaviour can resolve your issue, but please let us know if that is not the case.
Thanks!

**Comment by iheffernan** [comment]
Hey @ruisebas, thanks for responding. We have separate Compilation Conditions for Debug and Release and we are definitely using 'Release' when uploading to TestFlight. 

Additionally, I added this code snippet to my AppDelegate `didRegisterForRemoteNotificationsWithDeviceToken`...

```
#if DEBUG
print("in debug mode")
#else
print("in release mode")
#endif
```

When I change my Run scheme to our Release mode, I do see the `in release mode` message printed, but my Endpoint is still set to the APNS_SANDBOX Channel. 

Is it possible that Pinpoint is somehow caching my Endpoint profile? The Endpoint ID that gets created appears to be the same whenever it gets created for my device whether I install the app locally through Xcode or through TestFlight.

**Comment by ruisebas** [comment]
@iheffernan thanks for the update.
I will take a deeper look into this and post any updates in here.

Thanks!

**Comment by iheffernan** [comment]
One other not for comparision, here's a code snippet if I revert to v1.29.0 of the library...

```
let plugin = try Amplify.Analytics.getPlugin(for: "awsPinpointAnalyticsPlugin") as! AWSPinpointAnalyticsPlugin
            let escapeHatch = plugin.getEscapeHatch()
            
            #if DEBUG
            print("in debug mode")
            #else
            print("in release mode")
            #endif
            
            print("configuration.debug = \(escapeHatch.configuration.debug)")
```

I had to revert because the configuration isn't exposed in the escapeHatch in v2...

The `if DEBUG` block prints `in release mode`, but the escapeHatch for the plugin says `configuration.debug = true`. I stepped in a little deeper to the AWSPinpointAdapter:32 that has a similar `if DEBUG` and in _that_ code it actually went into the DEBUG block and set `pinpointConfiguration.debug = true`.

I'm not sure if v2 is the behaving the same way, but it looks like at my AppDelegate level it thinks DEBUG is off, but at the Pinpoint Configuration layer it thinks it is ON.

**Comment by iheffernan** [comment]
@ruisebas One more update, I was able to see the same thing with v2.5.0 of the library. In my AppDelegate the `if DEBUG` says I'm in Release mode, but in `AWSPinpointFactory.swift:33` a similar `#if DEBUG` sets `isDebug = true`. 

So in the current library, the low-level AWS client thinks we're in DEBUG mode, but the iOS app level thinks otherwise.

Let me know if there's anything else you'd like me to look at or anything else about our configuration that might help. Thanks...

**Comment by Montchat** [comment]
I am also experiencing this issue

**Comment by iheffernan** [comment]
@ruisebas If I could make a recommendation... Using the `DEBUG` flags to determine whether the environment should be SANDBOX or PRODUCTION does not align with how Apple allocates the device tokens. Apple uses the `APS Environment` entitlement value. So perhaps a more accurate way to set the ChannelType would be to parse the `embedded.mobileprovision` file and extract the `APS Environment` value and attempt to use that first. I've attached a file that would handle the parsing. Invocation would then look something like this...

**Comment by ruisebas** [comment]
Thank you for your suggestion and research. 
I'm still a bit puzzled as to what could be causing the discrepancy with the `DEBUG` flag.

I've made the changes to look for a provisioning profile first and check its APS Environment entitlement value, they are pushed to the [ruisebas/checking-provisioning-profile](https://github.com/aws-amplify/amplify-swift/tree/ruisebas/checking-provisioning-profile) branch. 

Would you be able to test if using that branch solves the issue for you?

Thanks!

**Comment by iheffernan** [comment]
@ruisebas That branch appears to work as expected with regards to setting the ChannelType. When debugging, I can see the new `ProvisioningProfileReader` hit and the ChannelType value for the created Endpoint seems to properly reflect the APS Environment when running on a physical device in Xcode or when the app is installed via Ad Hoc distribution.

One note - when I run in the iOS Simulator (I am on Apple Silicon, macOS 13.2.1, Xcode 14.2, and running on a simulated iPhone 14 Pro with iOS 16.2) it looks like it does hit the fallback logic to use the `DEBUG` flag. I guess the simulated device doesn't get an embedded provisioning file?

Let me know if you need anything else...

**Comment by ruisebas** [comment]
Sorry for the delay. The fix has been released as part of [2.7.0](https://github.com/aws-amplify/amplify-swift/releases/tag/2.7.0).

Please upgrade and let us know if you are still facing issues. Thanks!
